File: 000201-000400/000392.判断子序列/392.go
package main

import "fmt"

//方法一: 用时 0 ms
// 执行用时超过了 100 %
// 消耗内存超过了 71 %
func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}
	i, j := 0, 0
	for ; i < len(s) && j < len(t); {
		if s[i] == t[j] {
			i ++
		}
		j ++
	}
	return i == len(s)
}

func main ()  {
	fmt.Println(isSubsequence("abc", "ahbgdc"))
	fmt.Println(isSubsequence("axc", "ahbgdc"))
	fmt.Println(isSubsequence("", "ahbgdc"))
}

File: 000201-000400/000392.判断子序列/392.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
 //方法一: 用时 64 ms
 // 执行用时超过了 94 %
 // 消耗内存超过了 78 %
var isSubsequence = function(s, t) {
  let i = 0
  let j = 0
  while (i < s.length && j < t.length) {
    if (s[i] == t[j]) {
      i ++
    }
    j ++
  }
  return i == s.length
};

console.log(isSubsequence('abc', 'ahbgdc'))
console.log(isSubsequence('axc', 'ahbgdc'))
